### PR TITLE
Merge 'cardano-wallet-launcher' & 'cardano-wallet' together

### PR DESCRIPTION
**Pull request by KtorZ**
# Issue Number

#368 

# Overview

- [x] I have introduced a new 'launch' command to do what `cardano-wallet-launcher` used to do.
- [x] I have removed the 'cardano-wallet-launcher' executable and replaced occurences of it in the tests by `cardano-wallet launch`
- [x] I have renamed `cardano-wallet server` to `cardano-wallet serve` as discussed briefly on slack
- [x] I have enabled `--port` and `--random-port` options for the `launch` command
- [x] I have slightly adjusted launcher code to use the new logging infrastructure

# Comments

### CODE REVIEW

File: exe/wallet/Main.hs [hunk 1]
```diff
@@ -153,20 +172,26 @@ Usage:
...
   --port <INT>                port used for serving the wallet API [default: 8090]
   --random-port               serve wallet API on any available port (conflicts with --port)
-  --bridge-port <INT>         port used for communicating with the http-bridge [default: 8080]
-  --address-pool-gap <INT>    number of unused consecutive addresses to keep track of [default: 20]
   --size <INT>                number of mnemonic words to generate [default: 15]
-  --payment <PAYMENT>         address to send to and amount to send separated by @: '<amount>@<address>'
-  --network <STRING>          testnet or mainnet [default: testnet]
-  --database <FILE>           use this file for storing wallet state
   --state <STRING>            address state: either used or unused
+  --state-dir <DIR>           write wallet state (blockchain and database) to this directory
```
Comment (KtorZ) [inline exe/wallet/Main.hs:188]: I've sorted all the options alphabetically.

File: lib/core/test/integration/Test/Integration/Scenario/CLI/Mnemonics.hs [hunk 2]
```diff
@@ -36,24 +36,6 @@ spec = do
...
-        out `shouldContain` "cardano-wallet-launcher"
-        c `shouldBe` ExitFailure 1
-
-    describe "CLI_HELP - cardano-wallet-launcher shows help with" $  do
-        let test option = it option $ do
-                (Exit c, Stdout out) <- cardanoWalletLauncherCLI [option]
-                out `shouldContain` "cardano-wallet-launcher"
-                c `shouldBe` ExitSuccess
-        forM_ ["-h", "--help"] test
-
```
Comment (KtorZ) [inline lib/core/test/integration/Test/Integration/Scenario/CLI/Mnemonics.hs:56]: The same tests already exist for `cardano-wallet`

File: exe/wallet/Main.hs [hunk 3]
```diff
@@ -153,20 +172,26 @@ Usage:
...
 
 Examples:
+  # Launch and monitor a wallet backend and its associated node
+  cardano-wallet launch --network mainnet --random-port --state-dir .state-dir
+
+  # Start only a wallet server and connect to an already existing backend service
+  cardano-wallet serve --bridge-port 8080
```
Comment (piotr-iohk) [inline exe/wallet/Main.hs:195]: It might be a bit confusing if someone does not have a context (i.e. is `wallet backend` different than `wallet server`) I think I'd rephrase this so it is more clear, e.g.:

```
  # Launch and monitor a wallet server and its associated backend service
  cardano-wallet launch --network mainnet --random-port --state-dir .state-dir

  # Start only a wallet server and connect to an already existing backend service
  cardano-wallet serve --bridge-port 8080
```

**Review by Anviking — COMMENTED** [review]
lgtm based on a moderately quick look

**Review by Anviking — APPROVED** [review]
